src/util/simplify_expr.cpp

```diff
@@ -297,6 +297,78 @@ bool simplify_exprt::simplify_function_application(exprt &expr)
 
     return false;
   }
+  else if(func_id == ID_cprover_string_compare_to_func)
+  {
+    const refined_string_exprt &s1 =
+      to_string_expr(function_app.arguments().at(0));
+
+    const auto s1_data_opt = try_get_string_data_array(s1, ns);
+
+    if(!s1_data_opt)
+    {
+      return true;
+    }
+
+    const array_exprt &s1_data = s1_data_opt->get();
+
+    const refined_string_exprt &s2 =
+      to_string_expr(function_app.arguments().at(1));
+
+    const auto s2_data_opt = try_get_string_data_array(s2, ns);
+
+    if(!s2_data_opt)
+    {
+      return true;
+    }
+
+    const array_exprt &s2_data = s2_data_opt->get();
+
+    int res;
```

Comment:
Why is `res` declared here when only one branch below uses it?

Reply:
Refactored.